### PR TITLE
Move Resource to kernel

### LINKED ISSUE #1109: Move Resource to kernel
This also enables us to add `Concurrent#background`.

### PR DESCRIPTION
**Pull request by vasilmkd**
Resolves #1109.

### CODE REVIEW

**Review by RaasAhsan — APPROVED** [review]

File: core/shared/src/main/scala/cats/effect/LiftIO.scala [hunk 1]
```diff
@@ -135,4 +135,23 @@ object LiftIO {
...
+      def F1 = F10
+    }
+
+  abstract private class ResourceLiftIO[F[_]] extends LiftIO[Resource[F, *]] {
+    implicit protected def F0: LiftIO[F]
+    implicit protected def F1: Applicative[F]
+
+    def liftIO[A](ioa: IO[A]): Resource[F, A] =
+      Resource.liftF(F0.liftIO(ioa))
+  }
```
Comment (djspiewak) [inline core/shared/src/main/scala/cats/effect/LiftIO.scala:156]: As a random note, I kind of prefer putting instances on datatype companions whenever *possible*. So in this case, the instance would be on `Resource`'s companion (similar to how `Effect` has an instance on `IO`'s companion). Not a big deal; doesn't block the merge. Also curious as to others' thoughts.

Comment (vasilmkd) [reply core/shared/src/main/scala/cats/effect/LiftIO.scala:156]: That was the case. However, `LiftIO` doesn't exist in `kernel` and therefore it doesn't compile.

Comment (djspiewak) [reply core/shared/src/main/scala/cats/effect/LiftIO.scala:156]: Oh! Derp. That's obvious in retrospect. Okay cool. :-)

Comment (vasilmkd) [reply core/shared/src/main/scala/cats/effect/LiftIO.scala:156]: Rebase then?

Comment (djspiewak) [reply core/shared/src/main/scala/cats/effect/LiftIO.scala:156]: Merge please :-) Rebasing public history is usually not ideal.

Comment (vasilmkd) [reply core/shared/src/main/scala/cats/effect/LiftIO.scala:156]: Done.

**Pull request merged**